cmd/snap-update-ns/change.go

```diff
@@ -70,6 +74,29 @@ func (c *Change) Perform() error {
 		if err != nil {
 			return err
 		}
+		// If the mount point is not present then create a directory in its
+		// place.  This is very naive, doesn't handle read-only file systems
+		// but it is a good starting point for people working with things like
+		// $SNAP_DATA/subdirectory.
+		//
+		// We use lstat to ensure that we don't follow the symlink in case one
+		// was set up by the snap but this is not coded defensively as the code
+		// may execute concurrently with snap application processes.
+		//
+		// To have better defense we may consider freezing the freezer cgroup
+		// belonging to the snap during the execution of snap-update-ns.
+		if _, err := osLstat(c.Entry.Dir); err != nil {
+			if !os.IsNotExist(err) {
+				return err
+			}
+			// TODO: use the right mode and ownership.
+			if err := osMkdirAll(c.Entry.Dir, 0755); err != nil {
+				return err
+			}
+			if err := osChown(c.Entry.Dir, 0, 0); err != nil {
+				return err
+			}
```

Comment:
This chown is only doing the bottom directory-- if parents were also created, they would not be chowned. It is probably fine to chown all to root:root for now, but if we decide to allow arbitrary ownership, that will get tricky to limit it to only inside the snap and how to declare which directory is supposed to have which ownership (I'm not sure if that is planned).

Reply:
In thinking about this, I think that root:root 0755 is fine for all of the directories because the snap can always adjust the parent directories after creation. The mountpoint permissions will be covered up by whatever is mounted on it, and the snap could adjust that after the fact as well if desired (but, that will change the permissions of the original mountpoint-- we need to be really care of the mount options for things in SNAP_DATA, SNAP_COMMON or anywhere else the snaps can write to).

Reply:
I'll update the code to ensure we don't leave non root groups (as snap-confine is not setgid root) now.

Reply:
This is still a TODO